src/controllers/user.controller.ts

```diff
@@ -202,3 +297,116 @@ export const getFriends = async (req: Request, res: Response) => {
     return res.status(500).send({ message: error?.message });
   }
 };
+
+/**
+ * Sends a request to add a friend for the user with the specified userId.
+ * @param {Request} req - Express request object containing parameters: userId and friendId.
+ * @param {Response} res - Express response object used to send the response back to the client.
+ * @returns {Promise} Returns status code 200. Otherwise, returns an error status with an appropriate message.
+ */
+export const requestFriend = async (req: Request, res: Response) => {
+  const userId = req.params?.userId || '';
+  const friendId = req.query?.friendId?.toString() || '';
+
+  if (!ObjectId.isValid(userId)) {
+    return res.status(400).send({ message: 'Invalid user ID!' });
+  }
+
+  if (!ObjectId.isValid(friendId)) {
+    return res.status(400).send({ message: 'Invalid friend ID!' });
+  }
+
+  try {
+    const targetUser = await User.findByIdAndUpdate(friendId, {
+      $push: { friendRequests: userId },
+    });
+
+    if (targetUser === null) {
+      return res.status(400).send({ message: 'User does not exist!' });
+    }
+
+    return res
+      .status(200)
+      .send({ message: 'Successfully sent friend request!' });
```

Comment:
no check for whether `userId` exists? also no logic for actually "sending" the friend request?